### PR TITLE
Add EPSON RX8010SJ I2C RTC timer support

### PR DESCRIPTION
**Pull request by freakishness**
## Summary
Add EPSON RX8010SJ I2C RTC timer support
## Impact
None
## Testing
Test on imxrt1050-evk

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@freakishness please fix the style issue report here:
https://github.com/apache/incubator-nuttx/runs/5194562951?check_suite_focus=true
you can check locally by:
./tools/checkpatch.sh -G HEAD~...HEAD

**Comment by xiaoxiang781216** [comment]
@freakishness please squash your patchset into one, thanks.

File: drivers/timers/Kconfig [hunk 1]
```diff
@@ -291,6 +291,24 @@ config MCP794XX_I2C_FREQUENCY
 
 endif # RTC_MCP794XX
 
+config RTC_RX8010SJ
+  bool "RX8010SJ RTC Driver"
```
Comment (pkarashchenko) [inline drivers/timers/Kconfig:295]: please use TABs in Kconfig files

File: drivers/timers/rx8010.c [hunk 2]
```diff
@@ -0,0 +1,613 @@
...
+
+/****************************************************************************
+ * Name: dsxxxx_rtc_initialize
+ *
+ * Description:
+ *   Initialize the hardware RTC per the selected configuration.
+ *   This function is called once during the OS initialization sequence by
+ *   board-specific logic.
+ *
+ *   After dsxxxx_rtc_initialize() is called, the OS function
```
Comment (pkarashchenko) [inline drivers/timers/rx8010.c:193]:
```suggestion
 *   After rx8010_rtc_initialize() is called, the OS function
```

File: drivers/timers/rx8010.c [hunk 3]
```diff
@@ -0,0 +1,613 @@
...
+  int tens = ((int)value >> 4) * 10;
+  return tens + (value & 0x0f);
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: dsxxxx_rtc_initialize
```
Comment (pkarashchenko) [inline drivers/timers/rx8010.c:186]:
```suggestion
 * Name: rx8010_rtc_initialize
```

File: drivers/timers/rx8010.h [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+#define RX8010_FLAG_VLF		(1 << 1)
+#define RX8010_FLAG_AF		(1 << 3)
+#define RX8010_FLAG_TF		(1 << 4)
+#define RX8010_FLAG_UF		(1 << 5)
+
+#define RX8010_CTRL_AIE		BIT(3)
+#define RX8010_CTRL_UIE		BIT(5)
+#define RX8010_CTRL_STOP	BIT(6)
+#define RX8010_CTRL_TEST	BIT(7)
```
Comment (pkarashchenko) [inline drivers/timers/rx8010.h:72]: please align with others

File: drivers/timers/rx8010.h [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+#define RX8010_RESV31		0x31
+#define RX8010_IRQ		    0x32
+
+#define RX8010_TIME_SEC_BCDMASK 0x7f
+#define RX8010_TIME_MIN_BCDMASK 0x7f
+#define RX8010_TIME_HOUR24_BCDMASK 0x3f
+#define RX8010_TIME_DAY_MASK 0x07
+#define RX8010_TIME_DATE_BCDMASK 0x3f
+#define RX8010_TIME_MONTH_BCDMASK 0x1f
+#define RX8010_TIME_YEAR_BCDMASK 0xff
```
Comment (pkarashchenko) [inline drivers/timers/rx8010.h:60]: Please apply some spaces to align values

File: drivers/timers/rx8010.c [hunk 6]
```diff
@@ -0,0 +1,613 @@
...
+    {
+      ret = I2C_TRANSFER(g_rx8010.i2c, msg, 3);
+      if (ret < 0)
+        {
+          rtcerr("ERROR: I2C_TRANSFER failed: %d\n", ret);
+          return ret;
+        }
+    }
+  while ((buffer[1] & RX8010_TIME_SEC_BCDMASK) >
+  (seconds & RX8010_TIME_SEC_BCDMASK));
```
Comment (pkarashchenko) [inline drivers/timers/rx8010.c:486]:
```suggestion
  while ((buffer[1] & RX8010_TIME_SEC_BCDMASK) >
            (seconds & RX8010_TIME_SEC_BCDMASK));
```

File: drivers/timers/rx8010.c [hunk 7]
```diff
@@ -0,0 +1,613 @@
...
+
+  return OK;
+}
+
+static int rx8010_init()
```
Comment (pkarashchenko) [inline drivers/timers/rx8010.c:491]:
```suggestion
static int rx8010_init(void)
```

File: drivers/timers/rx8010.c [hunk 8]
```diff
@@ -0,0 +1,613 @@
...
+
+  /* set reserved register 0x31 with specified value of 0x08 */
+
+  buffer[0] = RX8010_RESV31;
+  buffer[1] = 0x08;
+  msg[0].frequency = CONFIG_RX8010SJ_I2C_FREQUENCY;
+  msg[0].addr      = RX8010_I2C_ADDRESS;
+  msg[0].flags     = 0;
+  msg[0].buffer    = buffer;
+  msg[0].length    = 2;
```
Comment (pkarashchenko) [inline drivers/timers/rx8010.c:539]:
```suggestion
```
I think most of `msg[0]` assignments can be done once. Only `msg[0].length` can be corrected before `I2C_TRANSFER`.

Does RX8010 support multi register access so all config can be written with a single I2C transaction?

File: drivers/timers/rx8010.c [hunk 9]
```diff
@@ -0,0 +1,613 @@
...
+  msg[1].length    = 1;
+
+  ret = I2C_TRANSFER(g_rx8010.i2c, msg, 2);
+  if (ret < 0)
+    {
+      rtcerr("ERROR: I2C_TRANSFER failed: %d\n", ret);
+      return ret;
+    }
+
+  if (buffer[0] & RX8010_FLAG_VLF)
```
Comment (pkarashchenko) [inline drivers/timers/rx8010.c:588]:
```suggestion
  if ((buffer[0] & RX8010_FLAG_VLF) != 0)
```

File: drivers/timers/rx8010.c [hunk 10]
```diff
@@ -0,0 +1,613 @@
...
+
+/* The state of the RX8010 chip.  Only a single RTC is supported */
+
+static struct rx8010_dev_s g_rx8010;
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static int rx8010_init();
```
Comment (pkarashchenko) [inline drivers/timers/rx8010.c:100]:
```suggestion
static int rx8010_init(void);
```

File: drivers/timers/rx8010.c [hunk 11]
```diff
@@ -0,0 +1,613 @@
...
+#ifndef CONFIG_RTC_DATETIME
+#  error CONFIG_RTC_DATETIME must be set to use this driver
+#endif
+
+#ifdef CONFIG_RTC_HIRES
+#  error CONFIG_RTC_HIRES must NOT be set with this driver
+#endif
+
+#ifndef CONFIG_RX8010SJ_I2C_FREQUENCY
+#  error CONFIG_RX8010SJ_I2C_FREQUENCY is not configured
```
Comment (pkarashchenko) [inline drivers/timers/rx8010.c:57]: `error` + `define` does not make sense. One has to be removed

**Comment by freakishness** [comment]
I'm sorry for causing unnecessary work to everyone because of my carelessness.I reworked the code as suggested and merged the commits

**Pull request merged**